clippy_lints/src/format_args.rs

```diff
@@ -559,6 +609,71 @@ impl<'tcx> FormatArgsExpr<'_, 'tcx> {
 
         false
     }
+
+    fn has_pointer_debug(&mut self, ty: Ty<'tcx>) -> bool {
+        let cx = self.cx;
+        let tcx = cx.tcx;
+        let typing_env = cx.typing_env();
+        let ty = tcx.normalize_erasing_regions(typing_env, ty);
+        if let Some(known) = self.has_pointer_format.get(&ty) {
+            return *known;
+        }
```

Comment:
This doesn't work since you only insert true for pointers, not types ultimately containing pointers.

Reply:
No, I insert true for the `ty`pe the function was called. The pointer type is in `next_ty`.